### PR TITLE
Update Fuchsia support for std::process. 

### PR DESCRIPTION
**Pull request by tedsta**
- Adds support for try_wait.
- Miscellaneous updates to keep up with Magenta changes.

I'll begin `sys/fuchsia` soon, just been bogged down with my actual job lately (I'm not on the Fuchsia team).

### CODE REVIEW

**Comment by raphlinus** [comment]
Looks good to me too, but I didn't do a careful review.

File: src/libstd/sys/unix/process/magenta.rs [hunk 1]
```diff
@@ -119,21 +119,14 @@ extern {
...
-// argument with each handle included in the process
-// arguments message.
-pub fn mx_hnd_info(hnd_type: u32, arg: u32) -> u32 {
-    (hnd_type & 0xFFFF) | ((arg & 0xFFFF) << 16)
-}
-
-extern {
-    pub fn mxio_get_startup_handle(id: u32) -> mx_handle_t;
+pub fn mx_job_default() -> mx_handle_t {
+    unsafe { return __magenta_job_default; }
```
Comment (solson) [inline src/libstd/sys/unix/process/magenta.rs]: Nit: Just use `unsafe { __magenta_job_default }`. Idiomatic Rust only uses `return` for early returns.

File: src/libstd/sys/unix/process/magenta.rs [hunk 2]
```diff
@@ -188,3 +181,124 @@ extern {
 
     pub fn launchpad_vmo_from_file(filename: *const c_char) -> mx_handle_t;
 }
+
+// Errors
+
+#[allow(unused)] pub const ERR_INTERNAL: mx_status_t = -1;
+
+// ERR_NOT_SUPPORTED: The operation is not implemented, supported,
```
Comment (solson) [inline src/libstd/sys/unix/process/magenta.rs]: This and all the following comments that document these constants should use the triple slash `///` which rustdoc recognizes as a doc comment and uses in the generated HTML.

File: src/libstd/sys/unix/process/process_fuchsia.rs [hunk 3]
```diff
@@ -164,6 +164,36 @@ impl Process {
...
+        let mut proc_info: mx_info_process_t = Default::default();
+        let mut actual: mx_size_t = 0;
+        let mut avail: mx_size_t = 0;
+
+        unsafe {
+            let status = mx_handle_wait_one(self.handle.raw(), MX_TASK_TERMINATED,
+                                            0, ptr::null_mut());
+            match status {
+                0 => { }, // Success
+                x if x == ERR_TIMED_OUT => {
```
Comment (solson) [inline src/libstd/sys/unix/process/process_fuchsia.rs]: I believe you can use constants directly as patterns, so this arm could be written as just `ERR_TIMED_OUT => {` with no `x if x ==`.

**Review by solson — COMMENTED** [review]
I know this is already merged, but I thought I'd leave a couple style notes for future reference.

**Pull request merged**